File: Deep-Learning-for-Beginners-master/Python/ch.2/TestDeltaBatch.py
import numpy as np
from Sigmoid import *
from DeltaBatch import *


def TestDeltaBatch():
    X = np.array([[0, 0, 1],
                  [0, 1, 1],
                  [1, 0, 1],
                  [1, 1, 1]])
    
    D = np.array([[0],
                  [0],
                  [1],
                  [1]])
        
    W = 2*np.random.random((1, 3)) - 1
        
    for _epoch in range(40000):
        W = DeltaBatch(W, X, D)
         
    N = 4
    for k in range(N):
        x = X[k,:].T
        v = np.matmul(W, x)
        y = Sigmoid(v)
        print(y)
        
if __name__ == '__main__':        
    TestDeltaBatch()

File: Deep-Learning-for-Beginners-master/ch.2/TestDeltaBatch.m
clear

X = [ 0 0 1;
      0 1 1;
      1 0 1;
      1 1 1;
    ];

D = [ 0
      0
      1
      1
    ];

[N,M]=size(X);  
W = 2*rand(1, M) - 1;

for epoch = 1:40000 
  W = DeltaBatch(W, X, D); 
end

for k = 1:N
  x = X(k, :)';
  v = W*x;
  y = Sigmoid(v);
  disp([' y_',num2str(k),' = ', num2str(y)])
end

